Sources/SourceKitLSP/Swift/DocumentFormatting.swift

```diff
@@ -137,26 +137,58 @@ private func edits(from original: DocumentSnapshot, to edited: String) -> [TextE
 extension SwiftLanguageService {
   package func documentFormatting(_ req: DocumentFormattingRequest) async throws -> [TextEdit]? {
     return try await format(
+      snapshot: documentManager.latestSnapshot(req.textDocument.uri),
       textDocument: req.textDocument,
       options: req.options
     )
   }
 
   package func documentRangeFormatting(_ req: DocumentRangeFormattingRequest) async throws -> [TextEdit]? {
     return try await format(
+      snapshot: documentManager.latestSnapshot(req.textDocument.uri),
       textDocument: req.textDocument,
       options: req.options,
       range: req.range
     )
   }
 
+  package func documentOnTypeFormatting(_ req: DocumentOnTypeFormattingRequest) async throws -> [TextEdit]? {
+    guard let server = self.sourceKitLSPServer else {
+      return nil
+    }
+    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+
+    let capabilities = await server.serverCapabilities(
+      for: capabilityRegistry.clientCapabilities,
+      registry: capabilityRegistry
+    )
+    guard let documentOnTypeFormattingProvider = capabilities.documentOnTypeFormattingProvider,
+      documentOnTypeFormattingProvider.firstTriggerCharacter == req.ch
+        || documentOnTypeFormattingProvider.moreTriggerCharacter?.contains(req.ch) == true,
+      let line = snapshot.lineTable.line(at: req.position.line),
+      /// No need to go through whitespace checking if the trigger is not a newline
+      !req.ch.isNewline || !line.allSatisfy(\.isWhitespace)
```

Comment:
Don't trigger formatting if we're in an empty line.
Otherwise will mess up writing code. You'd write `{}` and try to go into the braces to write more code, only for on-type formatting to immediately close the braces again.

Reply:
That’s a great explanation. Could you add it in the code?

Reply:
It appears this is no needed considering swift-format only will strictly format that single line the formatting is triggered on. 

I've removed the related code but kept the test because this problem can render on-type formatting useless if it happens in the future, as users will need to disable on-type formatting to be able to code at all.